### ISSUE TITLE
Consider setting up TerminalMenus

### ISSUE THREAD

**Issue by yeesian** [yeesian/ArchGDAL.jl#201]
I've been toying with the following idea of using https://docs.julialang.org/en/v1/stdlib/REPL/#TerminalMenus to create interactive prompts in the REPL, for example:

```julia
julia> ds = AG.read("ospy/data1/sites.shp")
GDAL Dataset (Driver: ESRI Shapefile/ESRI Shapefile)
File(s):
  ospy/data1/sites.shp
  ospy/data1/sites.shx
  ospy/data1/sites.dbf
  ospy/data1/sites.prj
  ospy/data1/sites.sbn
  ospy/data1/sites.sbx
  ...

Number of feature layers: 1
  Layer 0: sites (wkbPoint)


julia> AG.getlayer(ds)
Choose layer:
 > 1
Layer: sites
  Geometry 0 (): [wkbPoint], POINT (455552.418360...), ...
     Field 0 (ID): [OFTInteger], 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, ...
     Field 1 (COVER): [OFTString], shrubs, trees, rocks, grass, shrubs, ...
```

It can go much further than that, e.g. getting prompts on what are the methods that can be called on that dataset, before iteratively diving deeper that way. But before I go down that rabbit hole, curious to hear people's thoughts on the matter -- @visr @evetion @rafaqz.

**Comment by visr** [comment]
This looks really quite interesting, and appealing especially for exploratory / first time use. What about reproducability? Getting prompted while trying to run someones script could also be a bit annoying. Or would it be possible to convert the selection to code that gives the same result?

**Comment by yeesian** [comment]
> Or would it be possible to convert the selection to code that gives the same result?

Yeah, all great questions -- I'm thinking it might be good to have it in a separate/experimental package for that reason, so that it can have more "client-heavy types" to track the history of commands/options/etc.

**Comment by yeesian** [comment]
I'm also looking at https://github.com/mapbox/rasterio#rasterio-cli and https://github.com/Toblerity/Fiona#fiona-cli and the GDAL utilities for inspiration on what the user experience might be like first.

**Comment by yeesian** [comment]
I do think that the Julia REPL is where I'd want it to be -- since we'd get access to displays and help+shell modes and all the facilities that comes with Julialang packages that way.

**Comment by visr** [comment]
Yeah, I think that makes sense. If it is for REPL, does it also work for VSCode inline evaluation, Pluto, IJulia, e.g. all interactive environments?

**Comment by yeesian** [comment]
> If it is for REPL, does it also work for VSCode inline evaluation, Pluto, IJulia, e.g. all interactive environments?

I don't think TerminalMenus would work for those, that's a great point. I haven't thought through how such a solution should pair with packages in https://github.com/JuliaGizmos and be useful for e.g. https://jupyterlab.readthedocs.io/en/stable/.
